lib/cell.rb

```diff
@@ -0,0 +1,39 @@
+class Cell
+  attr_reader :coordinate, :ship #, :empty?
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @fired_upon = false
+  end
+
+  def empty?
+    return @ship == nil
+  end
+
+  def place_ship(ship)
+    if @ship == nil
+      @ship = ship
+    end
+  end
+
+  def fire_upon
+    @fired_upon = true
+    if @ship && !@fired_upon
```

Comment:
I think we need to rearrange this.  Line 21 will set the cell.fired_upon to "true".  So this condition will always fail on the 2nd condition !@fired_upon.